argparser.cpp

```diff
@@ -0,0 +1,129 @@
+#include <Windows.h>
+#include <tchar.h>
+#include "cpp_tstring.hpp"
+#include <iterator>
+#include <vector>
+#include "argparser.hpp"
+
+TCHAR helpMessage[] = TEXT("Usage: [program] [--output file] [--enable_cache cache_file] [--source source]\n\
+\t--no-splash - Don't show the FizzBuzz Enterprise Edition splash message.\n\
+\t--help - Display this help message, even if the input source is configured properly.\n\
+\t--output [file] - Specify the file to write the result to. If this argument is omitted, the result is written into the standard output\n\
+\t--enable_cache [cache_file] - Enable persistent caching.\n\
+\t--input_source [source] - Specify where to read the numbers from. If you don't specify this argument, the help message will be displayed instead. [source] can be one of:\n\
+\t\tstdinput - The program's standard input. It's recommended to disable the splash if you use this option when calling FizzBuzz Enterprise Edition from an another program.\n\
+\t\tfile filename.txt - A file. If you set the source to file, you need to specify the filename right after the source argument.\n\
+\t\ttestinput - A fake input source. It inputs all numbers from 1 to 100 (both sides included).\n");
+
+// See the help message above for more info about these flags.
+constexpr TCHAR HELP_FLAG[] = TEXT("--help");
```

Comment:
Would it make sense to add `static` here to avoid the external linkage? That should result in a smaller executable file.